### PR TITLE
[issue-593] add spdx2_document_from_scratch example

### LINKED ISSUE #593: Example missing imports
I just started evaluating this project for generating SPDX files manually, but had some trouble to finally generate a custom document:

* The example at https://github.com/spdx/tools-python#example omits all imports.
* The example starts with parsing an existing file. As documentation is sparse anyway, this requires me to do quite some extra research on how to create a document itself manually.
* Instead of exposing the corresponding classes from the submodules of `spdx.models` inside `spdx/models/__init__.py`, I am required to import them from the corresponding submodules, which seems like some unnecessary overhead.

### PR DESCRIPTION
**Pull request by armintaenzertng**
fixes #593

### CODE REVIEW

File: examples/spdx2_document_from_scratch.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+# Else, you can have a look at each entry's message and context
+# which will help you pinpoint the location of the invalidity.
+for message in validation_messages:
+    logging.warning(message.validation_message)
+    logging.warning(message.context.spdx_id)
+
+# Finally, we can serialize the document to any of the five supported formats.
+# Using the write_file() method from the write_anything module,
+# the format will be determined by the file ending: .spdx (tag-value), .json, .xml, .yaml. or .rdf (or .rdf.xml)
+write_file(document, "my_spdx_document.json")
```
Comment (stefan6419846) [inline examples/spdx2_document_from_scratch.py:145]: According to the specification, JSON serializations should use `*.spdx.json` (see https://spdx.github.io/spdx-spec/v2.3/conformance/#44-standard-data-format-requirements), so I would probably stick with the recommendation here as well?

File: examples/spdx2_document_from_scratch.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+# You can also add Annotations, Snippets and ExtractedLicensingInfo to the document in an analogous manner to the above.
+# Have a look at their respective classes if you are unsure about their properties.
+
+
+# This library provides comprehensive validation against the SPDX specification.
+# Note that details of the validation depend on the SPDX version of the document.
+validation_messages: List[ValidationMessage] = validate_full_spdx_document(document)
+
+# If the document is valid, validation_messages will be empty.
+assert validation_messages == []
```
[the remark this answers is not in the document]
Comment (armintaenzertng) [reply examples/spdx2_document_from_scratch.py:134]: You are right, that makes more sense

File: README.md [hunk 3]
```diff
@@ -94,7 +94,7 @@ instead of `bin`.
 
 ## Library usage
 1. **DATA MODEL**
-  * The `spdx_tools.spdx.model` package constitutes the internal SPDX v2.3 data model (v2.2 is a simply a subset of this). All relevant classes for SPDX document creation are exposed in the [__init__.py](src%2Fspdx_tools%2Fspdx%2Fmodel%2F__init__.py).
+  * The `spdx_tools.spdx.model` package constitutes the internal SPDX v2.3 data model (v2.2 is a simply a subset of this). All relevant classes for SPDX document creation are exposed in the `__init__.py` found [here](src%2Fspdx_tools%2Fspdx%2Fmodel%2F__init__.py).
```
Comment (meretp) [inline README.md:97]:
```suggestion
  * The `spdx_tools.spdx.model` package constitutes the internal SPDX v2.3 data model (v2.2 is simply a subset of this). All relevant classes for SPDX document creation are exposed in the `__init__.py` found [here](src%2Fspdx_tools%2Fspdx%2Fmodel%2F__init__.py).
```

Comment (armintaenzertng) [reply README.md:97]: fixed

**Review by meretp — APPROVED** [review]
LGTM

**Pull request merged**